inference-engine/src/gna_plugin/transformations/swap_input_matmul_gna.hpp

```diff
@@ -7,25 +7,168 @@
 #include <memory>
 #include <transformations_visibility.hpp>
 #include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ie/ie_common.h>
 
 namespace GNAPluginNS {
 
+class SwapInputMatMulFirstInputConstant;
+class SwapInputMatMulSecondInputConstant;
+class SwapInputMatMulWithBias;
+class SwapInputMatMulWithFq;
+
+namespace SwapInputMatMul {
+struct Helper {
+    static void SwapAndTransposeInputs(
+        std::shared_ptr<ngraph::opset8::MatMul> matmul_node,
+        std::shared_ptr<ngraph::Node> add,
+        std::shared_ptr<ngraph::Node> bias,
+        std::shared_ptr<ngraph::Node> fq,
+        const std::string& last_layer_name);
+    static std::shared_ptr<ngraph::Node> CreateMatmul(
+        bool is_first_constant,
+        ngraph::pattern::op::ValuePredicate const_predicate,
+        ngraph::pattern::op::ValuePredicate matmul_predicate = ngraph::pattern::has_static_shape());
+
+    template<typename T>
+    static bool CreateMatcher(std::shared_ptr<ngraph::pattern::Matcher>& matcher, ngraph::graph_rewrite_callback& callback) {
+        auto matmul1 = CreateMatmul(true, [](const ngraph::Output<ngraph::Node>& node) {
+            auto shape = node.get_node_shared_ptr()->get_output_shape(0);
+            return shape.size() == 2 && shape[0] > 8 && shape[0] % 8 == 0 && shape[1] % 8 == 0; });
```

Comment:
since we check the constant shape[0] % 8 == 0 && shape[1] % 8 == 0 doesn't matter, so this conditions can be removed
we can check the second input to know if we can transpose it: its one dimension must be <=8 and another one % 8 == 0
but probably we shouldn't check here if Transpose is supported since it can be removed in another transformation

Reply:
done